.gitconfig.khan

```diff
@@ -52,6 +52,9 @@
   # For after a commit fails with lint, and it says 'commit message saved...'
   recommit = "!git commit -F \"`git rev-parse --git-dir`\"/commit.save"
 
+  # For getting a list of commits that are on the current branch but not on the provided branch, but also copying them to the clipboard
+  clog = "!f() { git log \"$1\"..HEAD --oneline --pretty=format:\"%h %s\" --no-color | tee /dev/tty | pbcopy; }; f"
+
```

Comment:
I do wonder if there are any weird security issues with using the `tee /dev/tty` bit... 🤔

Reply:
I don't personally know!  Totally fine if you want to hold off on merging this to check in with the security team!

Reply:
Basically, I want the command to _both_ output to the terminal and copy it to the clipboard... when it was only sending it to the clipboard, it felt weird, because you also couldn't see what was getting sent.

The idea is that webapp folks could run this from the command-line on their deploy branches so they can easily copy the first line of their commit message so that they can paste it into Slack during a deploy without having stuff the extra tedious.

Reply:
> Basically, I want the command to both output to the terminal and copy it to the clipboard... when it was only sending it to the clipboard, it felt weird, because you also couldn't see what was getting sent.

Right, I got that.  But writing to /dev/tty is an unusual way to do that.  The right way to do it is to write to /dev/stdout.  But I don't know if /dev/stdout is an os x thing.  Though I think it is?

Since you are piping the output of stdout to somewhere else, though, you'll need to be careful about it.  I think you want to do something fancy, like this:
```
exec 3>&1; git log ... | tee /proc/self/fd/3 | pbpaste; exec 3>&-
```
What this does is copy stdout to fd 3 (so now fd 1 and fd3 both point to stdout).  Then you do git log and tee to fd 3, which means that tee is now writing to stdout twice, once on fd 1 and once on fd 3.  The one on fd1 goes to pbpaste due to the pipe.  The one on fd 3 stays going to stdout.  The final exec closes fd 3 (probably not necessary in this case, since I think this git command runs in a subshell.  But I'm not sure, so I'm doing it to be safe).

I think this can fail if fd 3 is already in use? Not an issue if you're in a subshell, but I don't know about otherwise.  But you can tree this and see how it works for you.

Reply:
(This assumes /proc/self/fd exists on os x.)

Reply:
@csilvers that's great feedback... I knew something felt off with just `dev/tty`. Seeing that @reidmit had a script, I just pivoted to this: https://github.com/Khan/our-lovely-cli/pull/839